core/src/main/java/org/pentaho/platform/util/Emailer.java

```diff
@@ -240,12 +246,11 @@ public boolean send() {
     try {
       // Get a Session object
       Session session;
-
-      if ( authenticate ) {
-        session = Session.getInstance( props, authenticator );
-      } else {
-        session = Session.getInstance( props );
+      IEmailConfiguration emailConfig = null;
+      if ( service != null ) {
+        emailConfig = service.getEmailConfig();
```

Comment:
This variable EmailConfig is never used until 
```
service.sendEmail( emailConfig, session, msg );
```
You should declare variables right before you use.

Currently the code flow is :

```
public boolean send() {
... ... ... 
      if ( service != null ) {
        emailConfig = service.getEmailConfig();
      }
... ... ...
if ( service != null ) {
        service.sendEmail( emailConfig, session, msg );
      }
      
... ... ...
return true
}

I would propose either approach below
[1]
`service.sendEmail( service.getEmailConfig(), session, msg );`
but then it asks why are we passing in a variable which the service already has access to so really
[2]
`service.sendEmail( session, msg );`
and then internally EmailService would call `getEmailConfig()`